go/pkg/client/cas.go

```diff
@@ -58,6 +59,7 @@ type uploadState struct {
 	// exactly once for a given state object (this is the whole point of the algorithm).
 	mu      sync.Mutex
 	clients []chan<- *uploadResponse
+	cancel  func()
```

Comment:
Can you update the comment above to indicate that mu now also protects the cancel function?

Reply:
Done.